runtime/wasm/src/asc_abi/mod.rs

```diff
@@ -23,10 +23,9 @@ compile_error!("big-endian targets are currently unsupported");
 /// The implementor must provide the direct Asc interface with `raw_new` and `get`.
 pub trait AscHeap: Sized {
     /// Allocate new space and write `bytes`, return the allocated address.
-    fn raw_new(&mut self, bytes: &[u8]) -> Result<u32, wasmi::Error>;
+    fn raw_new(&mut self, bytes: &[u8]) -> u32;
```

Comment:
The overall reasoning is:

1. The runtime deals with user provided pointers, assumes the types conform to a specific ABI and contain valid data (e.g. arrays have correct lengths). So it's "panic allowed" zone, so panics here don't become an attack vector.

2. Since panics are allowed anyways, we might as well simplify things by indexing into slices and unwrapping in places where an AS program would not cause trouble unless it's trying really hard.

Reply:
Taking into account the future necessity of separating deterministic from non-deterministic errors, panicking here is not what we want, but there are a lot of places to where panics are possible due to misbehaving AS mappings. So whether this PR makes the situation a bit better or a bit worse is not worth spending too much time on right now, we'll need to do a systematic pass at converting unwraps and index operators to `Result`s.